provisioning/Samples/service/ProvisioningAzureSasCredentialSample/Program.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Azure;
+using CommandLine;
+using Microsoft.Azure.Devices.Provisioning.Service;
+using System;
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ProvisioningAzureSasCredentialSample
+{
+    internal class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> parserResult = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            // Initialize SAS token credentials
+            Console.WriteLine("Creating sas credential.");
+
+            TimeSpan timeToLive = TimeSpan.FromHours(1);
+            DateTime expiresOn = DateTime.UtcNow.Add(timeToLive);
+            string sasToken = GenerateSasToken(parameters.HostName, parameters.SharedAccessKey, parameters.SharedAccessKeyName, expiresOn);
+            // Note: Pass the generated sasToken and not just the shared access signature when creating the AzureSasCredential.
+            AzureSasCredential sasCredential = new AzureSasCredential(sasToken);
+
+            // This is how the credential can be updated in the AzureSasCredential object whenever necessary.
+            // This sample just shows how to perform the update but it is not necessary to update the token
+            // until the token is close to its expiry.
+            DateTime newExpiresOn = DateTime.UtcNow.Add(timeToLive);
+            string updatedSasToken = GenerateSasToken(parameters.HostName, parameters.SharedAccessKey, parameters.SharedAccessKeyName, newExpiresOn);
+            sasCredential.Update(updatedSasToken);
```

Comment:
It wouldn't be too much to make a System.Timer run every hour and include the 4 lines of code needed to update it. Then you wouldn't have this repeated 2 sections, too.

Reply:
I added something simple for that. I've never worked with timers before so let me know if there's something I could do to improve the implementation.